cpp/src/arrow/CMakeLists.txt

```diff
@@ -541,8 +541,10 @@ if(ARROW_IPC)
 endif()
 
 if(ARROW_JSON)
+  add_definitions(-DARROW_WITH_JSON)
```

Comment:
No cmake expert here (and sorry for only asking now), so just wondering: can this also be "ARROW_JSON"? I see that for other cases, we reuse the same name as the cmake option

Reply:
This does seem redundant. I vaguely remember there was an issue with the minimal build at some point. Removing.